pkg/ccl/backupccl/restore_mid_schema_change_test.go

```diff
@@ -128,6 +128,9 @@ func expectedSCJobCount(scName string, isClusterRestore bool) int {
 		numBackgroundSCJobs = 2
 		// PK change + PK cleanup
 		expNumSCJobs = 2
+		if after {
+			expNumSCJobs = 1
```

Comment:
Adding to your explanation, since the validation fails in the after case (as you explained) we no longer expect the PK cleanup job to be created, and that's why we expect 1 job rather than 2